environment/environment.go

```diff
@@ -511,8 +544,21 @@ func (k8s *K8sManifest) parse(path string, obj interface{}, previousManifests ma
 	if err != nil {
 		return fmt.Errorf("failed to read k8s template file %s: %v", path, err)
 	}
+
+	chainlinkConnectDetails := struct {
```

Comment:
I suspect you won't like this, but other attempts at getting a clean marshall in just `previousManifests` were super messy. Kicked around your idea for utilizing callback functions a bit, but didn't get far.